commitment.go

```diff
@@ -26,14 +26,15 @@ type commitment struct {
 // newCommitment returns an commitment struct that notifies the provided
 // channel when log entries have been committed. A new commitment struct is
 // created each time this server becomes leader for a particular term.
-// 'voters' are the voting members of the cluster, including the
-// local server except when it's removed itself from the cluster.
+// 'configuration' is the servers in the cluster.
 // 'startIndex' is the first index created in this term (see
 // its description above).
-func newCommitment(commitCh chan struct{}, voters []string, startIndex uint64) *commitment {
-	matchIndexes := make(map[string]uint64, len(voters))
-	for _, voter := range voters {
-		matchIndexes[voter] = 0
+func newCommitment(commitCh chan struct{}, configuration Configuration, startIndex uint64) *commitment {
+	matchIndexes := make(map[string]uint64)
+	for _, server := range configuration.Servers {
+		if server.Suffrage == Voter {
```

Comment:
I wonder if it's worth adding a `CanVote()` member to `ServerSuffrage` in case we add other states later. Definitely not needed now, and probably a YAGNI, thinking out loud :-)